src/sentry/sentry_apps/token_exchange/refresher.py

```diff
@@ -58,7 +58,14 @@ def _validate(self) -> None:
         Validator(install=self.install, client_id=self.client_id, user=self.user).run()
 
         if self.token.application != self.application:
-            raise APIUnauthorized("Token does not belong to the application")
+            assert self.token.application is not None, "Application must exist on ApiToken"
+            raise SentryAppIntegratorError(
+                message="Token does not belong to the application",
+                webhook_context={
+                    "client_id_from_token": self.token.application.client_id[:4],
+                    "given_client_id": self.client_id[:4],
```

Comment:
too sus ?

Reply:
is it actionable if you only send the first few characters?

Reply:
probably not, we can give back the inputted client_id just fine but it's really sus to give back the working token one too. I'm wondering what other info we could give that would be helpful? Maybe the slug of the token owner ?

Reply:
For cases like https://github.com/getsentry/sentry/pull/83450/files#r1920807624, we could give the installation uuid of the given code and I think the installation for the mismatched token would be okay too. If someone owns the working `refresh_token` it would be safe to assume they have the corresponding integration installation too ?

Reply:
uuids sgtm